pkg/v2/service.go

```diff
@@ -343,6 +351,17 @@ func (s *service) Create(ctx context.Context, r *taskAPI.CreateTaskRequest) (_ *
 	// save the main task id and bundle to the shim for additional requests
 	s.id = r.ID
 	s.bundle = r.Bundle
+	pid := process.Pid()
+	if pid > 0 {
+		cg, err := cgroups.Load(cgroups.V1, cgroups.PidPath(pid))
+		if err != nil {
+			logrus.WithError(err).Errorf("loading cgroup for %d", pid)
+		}
+		s.cg = cg
```

Comment:
Can you call setCgroup from here to share the same code?

Reply:
setCgroup locks the same lock that this function does. Was trying to keep the same structure as runc's shim from containerd 1.2
https://github.com/containerd/containerd/blob/release/1.2/runtime/v2/runc/service.go#L307-L311